src/script/event/preprocessor/QuotedMessageMiddleware.ts

```diff
@@ -78,13 +78,13 @@ export class QuotedMessageMiddleware implements EventMiddleware {
     }
 
     this.logger.info(`Updating '${replies.length}' replies to updated message '${originalMessageId}'`);
-    replies.forEach(reply => {
+    replies.forEach(async reply => {
       const quote = reply.data.quote;
       if (quote && typeof quote !== 'string' && 'message_id' in quote && 'id' in event) {
         quote.message_id = event.id as string;
       }
       // we want to update the messages quoting the original message later, thus the timeout
-      setTimeout(() => this.eventService.replaceEvent(reply));
+      await this.eventService.replaceEvent(reply);
```

Comment:
The setTimeout can now be removed (we don't need to hackingly wait for the storage middleware to have done its job)